sample.bluetooth.peripheral_hr.frdm_kw41z_shield fails to build in frdm_mcxn947/mcxn947/cpu0 due to DTS issues

**Describe the bug**
sample.bluetooth.peripheral_hr.frdm_kw41z_shield fails to build in frdm_mcxn947/mcxn947/cpu0 due to DTS issues
It's expected any other test targeting this board which attempts to use the `arduino_serial` DT node will fail too.
The board twister yaml definition claims to support `arduino_serial` 

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=frdm_mcxn947/mcxn947/cpu0 ../samples/bluetooth/peripheral_hr/ -DSHIELD=frdm_kw41z 
3. See error
OR
1. twister -p frdm_mcxn947/mcxn947/cpu0 -s samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield

**Expected behavior**
No build failures or no claim of support

**Impact**
Blocking CI for PR which trigger this test

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/8293946654/job/22698629874?pr=70257#step:11:425
```
devicetree error: /__w/zephyr/zephyr/boards/shields/frdm_kw41z/frdm_kw41z.overlay:13 (column 1): parse error: undefined node label 'arduino_serial'
-- In: build/frdm_mcxn947_mcxn947_cpu0/samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield/zephyr, command: /usr/bin/python;zephyr/scripts/dts/gen_defines.py;--dts;build/frdm_mcxn947_mcxn947_cpu0/samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield/zephyr/zephyr.dts.pre;--dtc-flags;'';--bindings-dirs;/persistantzephyr/dts/bindings;--header-out;build/frdm_mcxn947_mcxn947_cpu0/samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield/zephyr/include/generated/devicetree_generated.h.new;--dts-out;build/frdm_mcxn947_mcxn947_cpu0/samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield/zephyr/zephyr.dts.new;--edt-pickle-out;build/frdm_mcxn947_mcxn947_cpu0/samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr.frdm_kw41z_shield/zephyr/edt.pickle;--edtlib-Werror;--vendor-prefixes;/persistantzephyr/dts/bindings/vendor-prefixes.txt
CMake Error at zephyr/cmake/modules/dts.cmake:283 (message):
  gen_defines.py failed with return code: 1
Call Stack (most recent call first):
  zephyr/cmake/modules/zephyr_default.cmake:132 (include)
  zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:5 (find_package)
```
**Environment (please complete the following information):**
 - OS: Linux (Ubuntu 22.04 / CI)
 - Toolchain : Host gcc (11 or 12)
 - Commit SHA or Version used : Today's main 42c8b7fb00a8f6343966e140b4bba593bb92cfdb

**Additional info**
Introduced in eaea4ec43e904b6180b61a78edd39daee26e4664

--- a/boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.yaml
+++ b/boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.yaml
@@ -15,6 +15,5 @@ toolchain:
   - gnuarmemb
   - xtools
 supported:
-  - arduino_serial
   - gpio
 vendor: nxp

